src/GafferScene/SceneReader.cpp

```diff
@@ -141,7 +163,13 @@ void SceneReader::hashBound( const ScenePath &path, const Gaffer::Context *conte
 {
 	SceneNode::hashBound( path, context, parent, h );
 
-	fileNamePlug()->hash( h );
+	ContextPtr tmpContext = new Context( *context, Context::Borrowed );
```

Comment:
I don't see anything wrong with the logic of what you have here, but I think it might make things a bit more readable to refactor this chunk into a `fileNameHash()` method up in the anonymous namespace, and reuse it below. How does that sound?